### PR TITLE
Add RIB icons for folder and rename intention

### PR DESCRIPTION
**Pull request by ShikaSD**
Updates the RIBs plugin to be able to 

1) Show ribs in the project pane with the icon;
2) Rename RIBs as a whole based on the name of the main interface.

### CODE REVIEW

File: android/tooling/rib-intellij-plugin/src/main/java/com/badoo/ribs/plugin/icons/RIBIconProvider.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        return null
+    }
+
+    private fun PsiElement.hasRib(): Boolean {
+//        // Check the cache for result
+//        getUserData(IS_RIB_KEY)?.let {
+//            return it
+//        }
```
Comment (zsoltk) [inline android/tooling/rib-intellij-plugin/src/main/java/com/badoo/ribs/plugin/icons/RIBIconProvider.kt]: Remove?

File: android/tooling/rib-intellij-plugin/src/main/java/com/badoo/ribs/plugin/icons/RIBIconProvider.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    fun KtSimpleNameExpression.isRib(): Boolean {
+        val references = references.map { it.resolve() }
+        return references.any { it is KtClass && it.fqName?.asString() == RIB_FQ_NAME }
+    }
+
+
+    companion object {
+        private const val RIB_FQ_NAME = "com.badoo.ribs.core.Rib"
```
Comment (zsoltk) [inline android/tooling/rib-intellij-plugin/src/main/java/com/badoo/ribs/plugin/icons/RIBIconProvider.kt]: I realise in Kotlin it's called fqname, let's be easier to understand here by writing RIB_FULLY_QUALIFIED_NAME?

File: android/tooling/rib-intellij-plugin/src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -36,6 +37,18 @@
                 <add-to-group anchor="last" group-id="NewGroup"/>
             </action>
         </group>
+        <group description="RIB architecture context menu" id="RibContextActionGroup" text="Rename RIB">
+            <action
+                class="com.badoo.ribs.plugin.action.RibRenameAction"
+                id="RenameRib"
+                text="Rename RIB...">
```
Comment (zsoltk) [inline android/tooling/rib-intellij-plugin/src/main/resources/META-INF/plugin.xml]: I love the idea btw!

**Comment by ShikaSD** [comment]
Removed the usages of `userData` in favour of more centralized and faster indexing, which also supports proper caching and invalidation out of the box.

**Pull request merged**